### PR TITLE
Add service monitoring

### PR DESCRIPTION
**Pull request by tudorraul**
This adds
- Sentry error monitoring for Salesforce hook
- metric for authentication failure (we expect all calls to have proper authentication cookies)
- metric for Salesforce updates, to get an understanding of how they change over time

### CODE REVIEW

**Comment by tudorraul** [comment]
@ostapneko

File: membership-attribute-service/app/actions/SalesforceAuthAction.scala [hunk 1]
```diff
@@ -2,16 +2,24 @@ package actions
...
 
   override def invokeBlock[A](request: Request[A], block: (Request[A]) => Future[Result]): Future[Result] = {
     val secretMatches = request.getQueryString(salesforceSecretParam).contains(Config.salesforceSecret)
-    if (secretMatches) block(request) else Future { unauthorized }
+
+    if (secretMatches) {
+      block(request)
+    }
+    else {
+      Logger.error(s"Invalid secret for request: '${request.uri}'")
```
Comment (ostapneko) [inline membership-attribute-service/app/actions/SalesforceAuthAction.scala]: Does it contain the query string?

File: membership-attribute-service/app/controllers/SalesforceHookController.scala [hunk 2]
```diff
@@ -13,10 +15,17 @@ import scala.concurrent.Future
...
 class SalesforceHookController @Inject() (attrService: AttributeService) {
+
+  val metrics = CloudWatch("SalesforceHookController")
+
   def createAttributes = SalesforceAuthAction.async(parse.xml) { request =>
     SFParser.parseOutboundMessage(request.body) match {
-      case \/-(attrs) => attrService.setAttributes(attrs).map(const(attrs))
-      case -\/(msg) => Future { ApiErrors.badRequest(msg) }
+      case \/-(attrs) =>
+        metrics.put("Update", 1)
```
Comment (ostapneko) [inline membership-attribute-service/app/controllers/SalesforceHookController.scala]: Will have to be updated after rebase to track deletion as well

File: membership-attribute-service/app/repositories/MembershipAttributesRepository.scala [hunk 3]
```diff
@@ -21,10 +21,14 @@ class MembershipAttributesRepository @Inject() (dynamo: AmazonDynamoDBScalaMappe
   def updateAttributes(attributes: MembershipAttributes): Future[Unit] =
     if (attributes.tier.isEmpty) {
       logger.debug(s"Delete attributes: $attributes")
-      dynamo.delete(attributes)
+      dynamo.delete(attributes).recover {
+        case t => Logger.error(s"Failed to delete attributes $attributes", t)
```
Comment (ostapneko) [inline membership-attribute-service/app/repositories/MembershipAttributesRepository.scala]: We want to rethrow the error

File: membership-attribute-service/app/repositories/MembershipAttributesRepository.scala [hunk 4]
```diff
@@ -21,10 +21,14 @@ class MembershipAttributesRepository @Inject() (dynamo: AmazonDynamoDBScalaMappe
...
+      }
     }
     else {
       logger.debug(s"Update attributes: $attributes")
-      dynamo.dump(attributes)
+      dynamo.dump(attributes).recover {
+        case t => Logger.error(s"Failed to update attributes $attributes", t)
```
Comment (ostapneko) [inline membership-attribute-service/app/repositories/MembershipAttributesRepository.scala]: cf supra

File: membership-attribute-service/app/actions/SalesforceAuthAction.scala (hunk 1 above)
Comment (tudorraul) [inline membership-attribute-service/app/actions/SalesforceAuthAction.scala]: yes, the query string with the invalid secret

**Comment by ostapneko** [comment]
:+1: After correcting last comment

**Pull request merged**